pkg/imagestore/imagestore.go

```diff
@@ -154,8 +178,50 @@ func (s *rhcosStore) Populate(ctx context.Context) error {
 	return errs.Wait()
 }
 
-func (s *rhcosStore) PathForParams(imageType, version, arch string) string {
-	return filepath.Join(s.dataDir, fmt.Sprintf("rhcos-%s-%s-%s.iso", imageType, version, arch))
+func (s *rhcosStore) PathForParams(imageType, openshiftVersion, arch string) string {
+	var version string
+	for _, entry := range s.versions {
+		if entry["openshift_version"] == openshiftVersion && entry["cpu_architecture"] == arch {
+			version = entry["version"]
+		}
+	}
+	return filepath.Join(s.dataDir, isoFileName(imageType, version, arch))
+}
+
+func isoFileName(imageType, version, arch string) string {
+	return fmt.Sprintf("rhcos-%s-%s-%s.iso", imageType, version, arch)
+}
+
+func (s *rhcosStore) cleanDataDir() error {
+	var expectedFiles []string
+	for _, version := range s.versions {
+		// Only add full isos here as we want to regenerate the minimal image on each deploy
+		expectedFiles = append(expectedFiles, isoFileName(ImageTypeFull, version["version"], version["cpu_architecture"]))
+	}
+
+	fileExpected := func(name string) bool {
+		for _, expectedFile := range expectedFiles {
+			if name == expectedFile {
+				return true
+			}
+		}
+		return false
+	}
+
+	dataDirFiles, err := os.ReadDir(s.dataDir)
+	if err != nil {
+		return err
+	}
+
+	for _, dataDirFile := range dataDirFiles {
+		if !fileExpected(dataDirFile.Name()) {
+			if err := os.RemoveAll(filepath.Join(s.dataDir, dataDirFile.Name())); err != nil {
+				return err
+			}
+		}
+	}
+
```

Comment:
maybe worth adding a log that folder was successfully cleaned, or/and a warning in case of partially cleaned.

Reply:
Added